### PR TITLE
[4.0] Clean up Bootstrap client side

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- No inline scripts
- A redo of #11922 (was RTC, before falling out of sync)

### Testing Instructions
Apply patch, ensure that tooltips, alerts, modals, tabs accordions etc still work

### Documentation Changes Required
Nope, internal changes

### CODE REVIEW

File: libraries/cms/html/bootstrap.php [hunk 1]
```diff
@@ -23,49 +23,6 @@
...
-	 *                                                               and left directions. To listen for a single direction, or multiple
-	 *                                                               unique offsets, just provide an object offset: { x: 10 }.
-	 *                                                               Use a function when you need to dynamically provide an offset
-	 *                                                               (useful for some responsive designs).
-	 *
-	 * @return  void
-	 *
-	 * @since   3.1
-	 */
-	public static function affix($selector = 'affix', $params = array())
```
Comment (mbabker) [inline libraries/cms/html/bootstrap.php]: ... deprecation?

File: libraries/cms/html/bootstrap.php [hunk 2]
```diff
@@ -77,22 +34,17 @@ public static function affix($selector = 'affix', $params = array())
...
 		// Only load once
-		if (isset(static::$loaded[__METHOD__][$selector]))
+		if (!empty(static::$loaded[__METHOD__][$selector]))
 		{
 			return;
 		}
 
 		// Include Bootstrap framework
-		JHtml::_('bootstrap.framework');
+		static::framework();
```
Comment (mbabker) [inline libraries/cms/html/bootstrap.php]: This actually needs to stay as it was so the JHtml override system continues to work correctly.

File: media/system/js/bootstrap-init.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+							$('.iframe').css('max-height', maxModalBodyHeight-modalBodyPadding);
+						}
+					} else {
+						if (modalHeight > maxModalHeight){
+							$('.modal-body').css({'max-height': maxModalBodyHeight, 'overflow-y': 'auto'});
+						}
+					}
+				}).on('hide.bs.modal', function () {
+					$('.modal-body').css({'max-height': 'initial', 'overflow-y': 'initial'});
+					// $('.modalTooltip').tooltip('destroy');
```
Comment (zero-24) [inline media/system/js/bootstrap-init.js]: is this expected to be just commeted out?

File: libraries/cms/html/bootstrap.php (hunk 2 above)
Comment (dgt41) [inline libraries/cms/html/bootstrap.php]: yup, copy paste from old branch

File: libraries/cms/html/bootstrap.php (hunk 1 above)
Comment (dgt41) [inline libraries/cms/html/bootstrap.php]: There is no affix in Bootstrap 4, so deprecating it won't work

Comment (mbabker) [inline libraries/cms/html/bootstrap.php]: It still needs to be deprecated in 3.x if it isn't already.

File: media/system/js/bootstrap-init.js (hunk 3 above)
Comment (dgt41) [inline media/system/js/bootstrap-init.js]: Bootstrap 4 is still alpha, so till they decide what will happen with tooltips and popovers this is good to be kept commented out...

Comment (C-Lodder) [inline media/system/js/bootstrap-init.js]: I'd suggest uncommenting it for now. I don't think the tooltip methods will change, and even in the rare chance they do, it's an easy fix.

Comment (dgt41) [inline media/system/js/bootstrap-init.js]: ok, done. Also merged the changes for the tooltip and the popover, this should be ready to go

**Comment by rjcf18** [comment]
I have tested this item :white_check_mark: successfully on e3fe25ce902d3fcde3666d8238197ecb0bfaf881<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on e3fe25ce902d3fcde3666d8238197ecb0bfaf881<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by dgt41** [comment]
@franz-wohlkoenig are the images in the correct order?

**Comment by franz-wohlkoenig** [comment]
@dgt41 Yes, applied PR having 4 Tabs instead 2 Tabs without PR.

**Comment by dgt41** [comment]
There was an bug with the tab titles. @franz-wohlkoenig can you check that is ok now?

**Comment by franz-wohlkoenig** [comment]
4 Tabs:
bildschirmfoto 2017-04-20 um 07 19 27

**Comment by Bakual** [comment]
> There was an bug with the tab titles.

@dgt41 Actually, the bug is that the tabs are doubled, the missing translated title wasn't the real bug :smile:

**Comment by dgt41** [comment]
Hmmm, I get 2 tabs here

**Comment by franz-wohlkoenig** [comment]
@dgt41 would it help to install fresh?

**Comment by dgt41** [comment]
Maybe, I still can't figure out why these tabs are bouble

**Comment by franz-wohlkoenig** [comment]
Fresh Install (staging), same Result:
bildschirmfoto 2017-04-20 um 08 39 55

**Comment by zero-24** [comment]
> Fresh Install (staging), same Result:

staging means 3.7? i guess you mean 4.0-dev? ;)

**Comment by dgt41** [comment]
Brand new install, fresh db, still can't replicate this (FF):
screen shot 2017-04-20 at 09 58 17

**Comment by dgt41** [comment]
@franz-wohlkoenig this is for J4, so @zero-24 is right!

**Comment by franz-wohlkoenig** [comment]
@zero-24 is right, i meant 4.0-dev.

**Comment by rjcf18** [comment]
Let me get a fresh installation from this again to verify this.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by rjcf18** [comment]
Weird, no tabs appear at all for me O.o<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by dgt41** [comment]
@rjcf18 are you getting any errors in the browsers console?

**Comment by rjcf18** [comment]
I've edited the comment, you can see the error in the previous comment.

**Comment by franz-wohlkoenig** [comment]
In FF-Console CSS:
Expected media feature name but found '-webkit-min-device-pixel-ratio'.  template.min.css:1:359
Expected colour but found 'auto'.  Expected colour but found '-webkit-focus-ring-color'.  Expected end of value but found '-webkit-focus-ring-color'.  Error in parsing value for 'outline'.  Declaration dropped.  template.min.css:1:449
Error in parsing value for '-webkit-text-size-adjust'.  Declaration dropped.  template.min.css:6:175
Unknown pseudo-class or pseudo-element '-webkit-inner-spin-button'.  Ruleset ignored due to bad selector.  template.min.css:6:1928
Unknown pseudo-class or pseudo-element '-webkit-search-cancel-button'.  Ruleset ignored due to bad selector.  template.min.css:6:2085
Unknown pseudo-class or pseudo-element '-webkit-file-upload-button'.  Ruleset ignored due to bad selector.  template.min.css:6:2181
Unknown property 'orphans'.  Declaration dropped.  template.min.css:6:2862
Unknown property 'widows'.  Declaration dropped.  template.min.css:6:2871
Expected colour but found 'auto'.  Expected colour but found '-webkit-focus-ring-color'.  Expected end of value but found '-webkit-focus-ring-color'.  Error in parsing value for 'outline'.  Declaration dropped.  template.min.css:6:4641
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:8643
Unknown pseudo-class or pseudo-element '-ms-expand'.  Ruleset ignored due to bad selector.  template.min.css:6:24921
Unknown pseudo-class or pseudo-element '-webkit-input-placeholder'.  Ruleset ignored due to bad selector.  template.min.css:6:25072
Unknown pseudo-class or pseudo-element '-ms-input-placeholder'.  Ruleset ignored due to bad selector.  template.min.css:6:25136
Unknown pseudo-class or pseudo-element '-ms-value'.  Ruleset ignored due to bad selector.  template.min.css:6:25459
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:30190
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:30386
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:30545
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:30978
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:31307
Unknown property 'user-select'.  Declaration dropped.  template.min.css:6:31809
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:40974
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:41652
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:43038
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:44475
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:44804
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:47625
Unknown property 'user-select'.  Declaration dropped.  template.min.css:6:48468
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:49533
Unknown pseudo-class or pseudo-element '-ms-value'.  Ruleset ignored due to bad selector.  template.min.css:6:50320
Unknown pseudo-class or pseudo-element '-ms-expand'.  Ruleset ignored due to bad selector.  template.min.css:6:50464
Unknown property 'user-select'.  Declaration dropped.  template.min.css:6:50987
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:51455
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:52669
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:52993
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:54268
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:54416
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:55055
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:55206
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:55848
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:55999
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:56643
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:56794
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:57385
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:57536
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:59839
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:63072
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:63164
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:63417
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:64976
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:68715
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:69230
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:69349
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:69817
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:74573
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:74939
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:75244
Unknown property 'line-break'.  Declaration dropped.  template.min.css:6:75893
Unknown property 'line-break'.  Declaration dropped.  template.min.css:6:77732
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:81486
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:82271
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:83387
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86082
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86150
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86493
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86564
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86908
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:86979
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:87323
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:87394
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:87739
Error in parsing value for 'display'.  Declaration dropped.  template.min.css:6:87810
Error in parsing value for 'position'.  Declaration dropped.  template.min.css:6:102802
Unknown property 'speak'.  Declaration dropped.  template.min.css:9:33914
Unknown property 'user-select'.  Declaration dropped.  template.min.css:9:34216
Error in parsing value for 'min-width'.  Declaration dropped.  template.min.css:9:36180
Error in parsing value for 'transition-delay'.  Declaration dropped.  template.min.css:9:45485
Error in parsing value for 'transition-delay'.  Declaration dropped.  template.min.css:9:48404

**Comment by dgt41** [comment]
@franz-wohlkoenig the CSS warnings/errors are not related to this PR

**Comment by C-Lodder** [comment]
No idea how on earth you're getting all these CSS errors. What version of FF are you using? And do you have any experimental features enabled?

**Comment by rjcf18** [comment]
On firefox the same happens:

deepinscreenshot20170420111717<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
@dgt41 CSS was the only Part with Errors.

@C-Lodder  Firefox 53.0 (64-bit), as i know i have no experimental Features enabled, only something like dict.cc

**Comment by dgt41** [comment]
@rjcf18 can you check once more, thanks!

**Comment by rjcf18** [comment]
Did another fresh installation, and I'm still having the same problem:

deepinscreenshot20170420114101

**Comment by rjcf18** [comment]
Oh wait wait sry, give me a minute let me try something

**Comment by rjcf18** [comment]
The HEAD was still at the previous commit :P  sry about that. But still the same problem though:

deepinscreenshot20170420115004

**Comment by dgt41** [comment]
are you using the browser in responsive mode?
I'm trying to figure out why I don't get this

**Comment by dgt41** [comment]
Still can't replicate this and without the ability to replicate it I can't fix it.
By the way are the tabs in the other views (e.g. article edit) working correctly?

**Comment by franz-wohlkoenig** [comment]
@dgt41 
bildschirmfoto 2017-04-20 um 13 29 44

**Comment by rjcf18** [comment]
In every other view I can see that they are working properly (buttons, accordeons, tabs, etc), I get that same error in the menu item type selection modal window despite it working well:

deepinscreenshot20170420122850

But I can't quite replicate what happens to @franz-wohlkoenig the tabs don't repeat for me:

deepinscreenshot20170420123315

deepinscreenshot20170420123344

**Comment by dgt41** [comment]
@franz-wohlkoenig I really don't know how you end up with double tabs, the code runs only once!
Ah, maybe the patch tester didn't delete some files?
Are you using git or patch tester here?

@rjcf18 same question are you using patch tester or git here?

**Comment by rjcf18** [comment]
I'm using git and for each of your commit I clone and install everything from scratch so there can be no problems with remainders from previous versions os some files.

**Comment by franz-wohlkoenig** [comment]
@dgt41 i use Patchtester. Will reinstall it.

**Comment by franz-wohlkoenig** [comment]
Issue stay same after reinstalling Patchtester; same in Chrome 57, Safari 10.1.

**Comment by dgt41** [comment]
@franz-wohlkoenig I'm afraid this PR cannot be tested with patchtester

**Comment by franz-wohlkoenig** [comment]
@dgt41 okay. Possible to write this in future Test instructions?

**Comment by dgt41** [comment]
@franz-wohlkoenig I think there is a known limitation for patch tester some files, @mbabker can explain the issue better

**Comment by franz-wohlkoenig** [comment]
@dgt41 theres also a Comment that Patchtester in J4 is experimental – but most works. I thought you know that this PR isn't testable by Patchtester and so to write info.

**Comment by mbabker** [comment]
Unless something changed with GitHub this PR is fine with patch tester.  The biggest problems relate to binary files, none of the files changed here fall into that category.

The patch tester quirk is it does not patch files.  It does a full replace based on the file from the remote branch.  So if that is in any way out of sync with the base branch, it can cause problems.

**Comment by dgt41** [comment]
@franz-wohlkoenig honestly I didn't even think about patchtester in the first place. Till J4 repo become public all the participants were using git.

**Comment by Bakual** [comment]
Please also check that you browser cache is cleared before testing. Not that it caches some parts of the JS.
When using Chrome, a good way to have cache always cleared is to have the development tools open (F12) and have "Disable cache (while DevTools is open)" enabled in its settings.

**Comment by rjcf18** [comment]
Yep I was aware of that option and have it activated, plus I clear the cache anyways just in case both in chrome and firefox. For this case I guess the problem isn't related to that, but yeah it's a good practice.

**Comment by dgt41** [comment]
@Bakual thanks, should be fine now

**Comment by rjcf18** [comment]
I have tested this item :white_check_mark: successfully on 1d60931586fde7f330fef8598c068bc8459f40bc<br /><br />The problem seems to be solved, no longer getting the error on the console on the modal window and the tabs in Joomla! Update seem to be displayed correctly and with no errors:

deepinscreenshot20170420234148

deepinscreenshot20170420233638
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on 1d60931586fde7f330fef8598c068bc8459f40bc<br /><br />Test on fresh Install 4.0.0-dev, double Tabs Issue stay:<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by dgt41** [comment]
@franz-wohlkoenig I think patchetester is not behaving with this issue

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on 1d60931586fde7f330fef8598c068bc8459f40bc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15374">issues.joomla.org/tracker/joomla-cms/15374</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**